controllers/admin/config.go

```diff
@@ -293,6 +293,12 @@ func SetWebServerPort(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// don't panic when input is not a float
+	_, is_float := configValue.Value.(float64)
+	if !is_float {
+		return
+	}
```

Comment:
Thanks for adding this check!  Instead of asserting the type once here to check it, and then another time to actually use it, could this be swaped around a bit  (excuse the bad code inside a comment box):

```
if port, ok := configValue.Value.(float64); ok {
...setHttpPort(port)
} else {
BadRequestHandler(w, "bad request or some message")
}
```

And the same can probably be done with the `SetWebServerIP` check/use as well.

Reply:
Gotcha, will do.

Reply:
71aca7e should have a better logic and also checks port range and validates given IP